### PR TITLE
Added toggle to example section IMPORT/EXPORT for cloud storage providers v21.1

### LINKED ISSUE #4398: backup/restore/import: Provide example for AWS_ENDPOINT parameter
Exalate commented:

We provide an AWS_ENDPOINT parameter in our cloud storage code (used by backup/restore/import/CDC)  which allows CockroachDB to work with HTTP(S) servers that implement the S3 API. We should provide an example of how this works.

For reference, if you have an S3-alike system running at `https://s3-alike.cockroach.industries:9099`, and you have permission to write to bucket `testbucket` with access key `foo` and secret key `bar`, you would write a backup statement with a cloud storage URL like:

`BACKUP DATABASE whatever TO 's3://testbucket/backup/path?AWS_ACCESS_KEY_ID=foo&AWS_SECRET_ACCESS_KEY=bar&AWS_ENDPOINT=https://s3-alike.cockroach.industries:9099';`

Jira Issue: DOC-244

### PR DESCRIPTION
**Pull request by kathancox**
Relates to #10544, #10770 

Like `BACKUP`/`RESTORE` 21.1 have changed the example section for both `IMPORT` and `EXPORT` to toggle for cloud provider.

### CODE REVIEW

**Comment by kathancox** [comment]
As a note: links that build is failing on are dependent on another branch/PR — will fix with other feedback corrections.

**Review by mwang1026 — COMMENTED** [review]
I looked overall it looks good. It's really hard to review this one in particular by the code, so I'll give it another deeper read when there's a preview available

> Reviewed 1 of 2 files at r1, 3 of 5 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10792#-:-MdNe8nB5eudztiF7-Ay:b-604e6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox)

**Review by mwang1026 — APPROVED** [review]
lgtm!

I think Ian had a separate thread about how it could be confusing to users if we just have cloud storage examples but not userfile, etc. that I think we can address in the other issue. We could also make a small change here to say that we have cloud example but that they follow similar pattern for userfile, etc. but I'll leave that up to you re: when / how to handle the disambiguation of this stuff

> Reviewed 8 of 8 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10792#-:-Me_oJjc0Ki_LpWaH8cN:bkowzdi)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox)

___
*[v21.1/use-cloud-storage-for-bulk-operations.md, line 160 at r4](https://reviewable.io/reviews/cockroachdb/docs/10792#-Me_oCuQ7RAqufDJtlxO:-Me_oCuQ7RAqufDJtlxP:b8ozsio) ([raw file](https://github.com/cockroachdb/docs/blob/ff922cf938f721955424f867b0957c5787049d3e/v21.1/use-cloud-storage-for-bulk-operations.md#L160)):*
> ```
> .s
> ```
weird typo? :D

**Comment by kathancox** [comment]
Ha, oh boy... guess I didn't quite hit `CMD` when trying to save with that `s`. 

Re the userfile examples, yes, once I have sorted out the userfile examples I will work out how to add that here.

File: v21.1/export.md [hunk 1]
```diff
@@ -114,6 +130,106 @@ export16808a04292505c80000000000000001-n1.0.csv.gz |   17 |   824
...
+### Export using a `SELECT` statement
+
+{% include copy-clipboard.html %}
+~~~ sql
+> EXPORT INTO CSV
+  'azure://{CONTAINER NAME}/{customer-export-data}?AZURE_ACCOUNT_NAME={ACCOUNT NAME}&AZURE_ACCOUNT_KEY={ENCODED KEY}'
+  FROM SELECT * FROM bank.customers WHERE id >= 100;
+~~~
+
+### Non-distributed export using the SQL shell
```
Comment (rmloveland) [inline v21.1/export.md:157]: same re: client?

File: v21.1/export.md [hunk 2]
```diff
@@ -73,12 +73,28 @@ Successful `EXPORT` returns a table of (perhaps multiple) files to which the dat
...
+
+<div class="filters clearfix">
+  <button class="filter-button" data-scope="s3">Amazon S3</button>
+  <button class="filter-button" data-scope="azure">Azure Storage</button>  
+  <button class="filter-button" data-scope="gcs">Google Cloud Storage</button>
+</div>
+
+<section class="filter-content" markdown="1" data-scope="s3">
+
+{{site.data.alerts.callout_info}}
```
Comment (rmloveland) [inline v21.1/export.md:86]: same re: include?

File: v21.1/export.md [hunk 3]
```diff
@@ -114,6 +130,106 @@ export16808a04292505c80000000000000001-n1.0.csv.gz |   17 |   824
...
+### Export a table
+
+{% include copy-clipboard.html %}
+~~~ sql
+> EXPORT INTO CSV
+  'azure://{CONTAINER NAME}/{customer-export-data}?AZURE_ACCOUNT_NAME={ACCOUNT NAME}&AZURE_ACCOUNT_KEY={ENCODED KEY}'
+  WITH delimiter = '|' FROM TABLE bank.customers;
+~~~
+
+### Export using a `SELECT` statement
```
Comment (rmloveland) [inline v21.1/export.md:148]: same re: link to "selection queries" ?

File: v21.1/export.md [hunk 4]
```diff
@@ -114,6 +130,106 @@ export16808a04292505c80000000000000001-n1.0.csv.gz |   17 |   824
...
+export16808a04292505c80000000000000001-n1.0.csv.gz |   17 |   824
+(1 row)
+~~~
+
+</section>
+
+<section class="filter-content" markdown="1" data-scope="gcs">
+
+{{site.data.alerts.callout_info}}
+The examples in this section use the `AUTH=specified` parameter, which will be the default behavior in v21.2 and beyond for connecting to Google Cloud Storage. For more detail on how to pass your Google Cloud Storage credentials with this parameter, or, how to use `implicit` authentication, read [Use Cloud Storage for Bulk Operations — Authentication](use-cloud-storage-for-bulk-operations.html#authentication).
```
Comment (rmloveland) [inline v21.1/export.md:185]: include?

File: v21.1/export.md [hunk 5]
```diff
@@ -114,6 +130,106 @@ export16808a04292505c80000000000000001-n1.0.csv.gz |   17 |   824
...
+### Export using a `SELECT` statement
+
+{% include copy-clipboard.html %}
+~~~ sql
+> EXPORT INTO CSV
+  'gs://{BUCKET NAME}/{customer-export-data}?AUTH=specified&CREDENTIALS={ENCODED KEY}'
+  FROM SELECT * FROM bank.customers WHERE id >= 100;
+~~~
+
+### Non-distributed export using the SQL shell
```
Comment (rmloveland) [inline v21.1/export.md:208]: same re: client?

File: v21.1/import-into.md [hunk 6]
```diff
@@ -162,167 +162,206 @@ Pausing and then resuming an `IMPORT INTO` job will cause it to restart from the
...
+<div class="filters clearfix">
+  <button class="filter-button" data-scope="s3">Amazon S3</button>
+  <button class="filter-button" data-scope="azure">Azure Storage</button>
+  <button class="filter-button" data-scope="gcs">Google Cloud Storage</button>
+</div>
 
-Amazon S3:
+<section class="filter-content" markdown="1" data-scope="s3">
+
+{{site.data.alerts.callout_info}}
```
Comment (rmloveland) [inline v21.1/import-into.md:175]: same re: include?

File: v21.1/import-into.md [hunk 7]
```diff
@@ -162,167 +162,206 @@ Pausing and then resuming an `IMPORT INTO` job will cause it to restart from the
...
-      'azure://acme-co/customer-import-data1.3.csv?AZURE_ACCOUNT_KEY=hash&AZURE_ACCOUNT_NAME=acme-co',
-      'azure://acme-co/customer-import-data1.4.csv?AZURE_ACCOUNT_KEY=hash&AZURE_ACCOUNT_NAME=acme-co',
-      'azure://acme-co/customer-import-data1.5.csv?AZURE_ACCOUNT_KEY=hash&AZURE_ACCOUNT_NAME=acme-co',    
+      'azure://{CONTAINER NAME}/{customers.csv}?AZURE_ACCOUNT_NAME={ACCOUNT NAME}&AZURE_ACCOUNT_KEY={ENCODED KEY}'
     );
 ~~~
 
-Google Cloud:
+{{site.data.alerts.callout_info}}
+The column order in your statement must match the column order in the CSV being imported, regardless of the order in the existing table's schema.
```
Comment (rmloveland) [inline v21.1/import-into.md:251]: same re: " .. IMPORT statement"?

File: v21.1/import-into.md [hunk 8]
```diff
@@ -162,167 +162,206 @@ Pausing and then resuming an `IMPORT INTO` job will cause it to restart from the
...
+      fields_enclosed_by='"',
+      fields_escaped_by='\';
 ~~~
 
-Google Cloud:
+</section>
+
+<section class="filter-content" markdown="1" data-scope="gcs">
+
+{{site.data.alerts.callout_info}}
```
Comment (rmloveland) [inline v21.1/import-into.md:302]: same re: include?

File: v21.1/import-into.md [hunk 9]
```diff
@@ -162,167 +162,206 @@ Pausing and then resuming an `IMPORT INTO` job will cause it to restart from the
...
+    CSV DATA (
+      'gs://{BUCKET NAME}/{customers.csv}?AUTH=specified&CREDENTIALS={ENCODED KEY}'
     );
 ~~~
 
-For more detailed information about importing data from Avro and examples, see [Migrate from Avro](migrate-from-avro.html).
-
-### Import into an existing table from a delimited data file
+{{site.data.alerts.callout_info}}
+The column order in your statement must match the column order in the CSV being imported, regardless of the order in the existing table's schema.
```
Comment (rmloveland) [inline v21.1/import-into.md:317]: same re: IMPORT statement ?

**Review by rmloveland — APPROVED** [review]
LGTM with various small comments

File: v20.2/import.md [hunk 10]
```diff
@@ -288,16 +526,25 @@ Azure:
...
-    'azure://acme-co/customer-import-data1.3.csv?AZURE_ACCOUNT_KEY=hash&AZURE_ACCOUNT_NAME=acme-co',
-    'azure://acme-co/customer-import-data1.4.csv?AZURE_ACCOUNT_KEY=hash&AZURE_ACCOUNT_NAME=acme-co',
-    'azure://acme-co/customer-import-data1.5.csv?AZURE_ACCOUNT_KEY=hash&AZURE_ACCOUNT_NAME=acme-co',    
-);
+CSV DATA ('azure://{CONTAINER NAME}/{customer-import-data.csv}?AZURE_ACCOUNT_NAME={ACCOUNT NAME}&AZURE_ACCOUNT_KEY={ENCODED KEY}')
+;
 ~~~
 
-Google Cloud:
+To use a file to specify the table schema:
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply v20.2/import.md:533]: PS this comment above was not meant as anything for you to address in this PR Kathryn!  just typing out some thoughts about more like team-level stuff

[the remark this answers is not in the document]
Comment (kathancox) [reply v20.2/import.md:533]: I agree though! I don't fully understand the liquid templating logic (or know it), so would love to discuss what would be best here. Having now worked on a couple of PRs involving updates to the toggled sections/details, it is a pain. If there is a way to create logic, pulling in the different cloud connection strings, I would at least like to know what that would look like/pros & cons.

[the remark this answers is not in the document]
Comment (kathancox) [reply v20.2/import.md:533]: Like... is there a way to pull one set of examples into an _includes, set a marker where the connection string is, and then have logic that's like `if under this section then aws string` etc.

[the remark this answers is not in the document]
Comment (rmloveland) [reply v20.2/import.md:537]: > Like... is there a way to pull one set of examples into an _includes, set a marker where the connection string is, and then have logic that's like if under this section then aws string et

idk (I also do not know how to use the templating logic stuff in any serious way, I'm merely aware it exists) but I bet our team will have to find out together at some point in the near to mid term :-)

**Pull request merged**